### ISSUE TITLE
[10.0][11.0][12.0] negative order line without tax, generates refund even when total amount of the order is positive

### ISSUE THREAD

**Issue by yung-wang** [odoo/odoo#26726]
Impacted versions:
[10.0][11.0][12.0]

Steps to reproduce:
Create a sales order and add the following lines
Serviceproduct Warranty - qty 1 - price 600.00 with tax 15% (total = 690.00)
Serviceproduct Cashback - qty 1 - price -670.00 without tax
total order amount = +20.00

when you create an invoice, the system will create a refund with total invoice amount -20.00 [NOT OK]
The invoice total of the sales order is positive, so it should create an invoice instead refund.
I try the same with a deposit of 670,- without tax and the same happens.

I think odoo looks at the base amount, if that is negative it will create a refund.

Expected behavior:
Based on the total to invoice amount, the system should create an invoice if positive and if negative a refund.

Video/Screenshot link (optional):

**Comment by sswapnesh** [comment]
> I think odoo looks at the base amount, if that is negative it will create a refund.
or May be odoo looks at Total amount before computing taxes.

**Comment by sswapnesh** [comment]
However, Initially odoo creates Customer Invoice but later `Invoice Type` is being changed to `Credit Note`

![](https://image.prntscr.com/image/_1FY76_8RpO_30VKKfS5oQ.png)

**Comment by sswapnesh** [comment]
Here is the Condition from [this](https://github.com/odoo/odoo/commit/94716a3f14d929574c46ef165e4364d12a5903ed#diff-3de7d15f09aac4a2e96ae7740b24a100R270) Commit.
https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L445
```
if invoice.amount_untaxed < 0:
                invoice.type = 'out_refund'
```

**Comment by pedrobaeza** [comment]
Maybe the condition can be changed to `amount_taxed` instead. @qdp-odoo what do you think about this?

**Comment by sswapnesh** [comment]
@pedrobaeza  I believe you meant `amount_total`, There is not field named `amount_taxed`.

**Comment by pedrobaeza** [comment]
Yeah, right, I remember bad the name of the field. Comment edited.

**Comment by qdp-odoo** [comment]
looks like a good idea

**Comment by yung-wang** [comment]
Hey guys, any idea when this will be solved? Seems like a very small change in the code. Of course the impact could be bigger. Do we need functional review?

p.s. This is also still present in V12

**Comment by sswapnesh** [comment]
@Yenthe666 
I had created Branch with Fix in v10 but didn't created PR as I was not sure where to target. (Maybe v12 for now)

**Comment by yung-wang** [comment]
Sorry if I don't understand the PR process.
Please fix in all versions if possible. Why skip 11?

**Comment by Yenthe666** [comment]
@yung-wang if the PR is accepted and merged in for 10.0 it will automatically be added to 11.0 & 12.0, fixes always go upward.
@sswapnesh don't worry, I made one against 10.0 and have a closed one against 12.0 if Quentin would like that more.

**Comment by yung-wang** [comment]
thanks for the explanation!

**Comment by Yenthe666** [comment]
@amkarthik you could trigger the functions from a scheduled action, XML-RPC or a part of custom code.

**Comment by sswapnesh** [comment]
@Yenthe666 I don't think this would work as they are already Processed. ORM will not allow to do any operation on that(Specially paid Once),

So IMO solution would be create Invoice of Refund amount and than create new Invoice with same details. (Duplicate won't work as invoice type is refund already)

**Comment by Yenthe666** [comment]
Euhh that wouldn't be very good / neat from an accounting point of perspective I guess.

**Comment by sswapnesh** [comment]
True, But could not guess any alternative option.
